### PR TITLE
feat(wallet-dashboard): style staked vesting overview

### LINKED ISSUE #4260: Style staked vesting overview
Style staked vesting overview:
- total stake
- total earned
- list of stakes (note these are grouped, they should be groupedalready in the codebase)

Image

### PR DESCRIPTION
**Pull request by panteleymonchuk**
# Description of change

Styling for vesting

## Links to any relevant issues

Closes #4260

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by evavirseda — CHANGES_REQUESTED** [review]
ci is failing :pray:

File: apps/wallet-dashboard/app/(protected)/vesting/page.tsx [hunk 1]
```diff
@@ -62,6 +81,33 @@ function VestingDashboardPage(): JSX.Element {
...
+        IOTA_TYPE_ARG,
+    );
+    const [totalLockedFormatted, totalLockedSymbol] = useFormatCoin(
+        vestingSchedule.totalLocked,
+        IOTA_TYPE_ARG,
+    );
+    const [availableClaimingFormatted, availableClaimingSymbol] = useFormatCoin(
+        vestingSchedule.availableClaiming,
+        IOTA_TYPE_ARG,
+    );
```
Comment (evavirseda) [inline apps/wallet-dashboard/app/(protected)/vesting/page.tsx]: this is not part of this issue, it's done in another pr :see_no_evil:

File: apps/wallet-dashboard/app/(protected)/vesting/page.tsx [hunk 2]
```diff
@@ -142,81 +188,155 @@ function VestingDashboardPage(): JSX.Element {
...
+                                    isTextTruncated
+                                />
+                                <CardAction
+                                    buttonType={ButtonType.Primary}
+                                    type={CardActionType.Button}
+                                    title="Collect"
+                                    onClick={handleCollect}
+                                />
+                            </Card>
+                        ) : null}
```
Comment (evavirseda) [inline apps/wallet-dashboard/app/(protected)/vesting/page.tsx]: same with all of this :see_no_evil: :sweat_smile:

File: apps/wallet-dashboard/app/(protected)/vesting/page.tsx [hunk 3]
```diff
@@ -44,6 +62,7 @@ function VestingDashboardPage(): JSX.Element {
     const { data: timelockedStakedObjects } = useGetTimelockedStakedObjects(account?.address || '');
     const { mutateAsync: signAndExecuteTransaction } = useSignAndExecuteTransaction();
 
+    console.log('dd', timelockedStakedObjects);
```
Comment (evavirseda) [inline apps/wallet-dashboard/app/(protected)/vesting/page.tsx]: :broom:

File: apps/wallet-dashboard/app/(protected)/vesting/page.tsx [hunk 4]
```diff
@@ -142,81 +188,155 @@ function VestingDashboardPage(): JSX.Element {
...
+            <Panel>
+                <Title title="Staked Vesting" />
+                <div className="flex flex-col px-lg py-sm">
+                    <div className="flex flex-row gap-md">
+                        <DisplayStats
+                            label="Your stake"
+                            value={`${totalStakedFormatted} ${totalStakedSymbol}`}
+                        />
+                        <DisplayStats
+                            label="Total Unlocked"
```
Comment (evavirseda) [inline apps/wallet-dashboard/app/(protected)/vesting/page.tsx]: this should be `Earned` not the `Total Unlocked`
image

File: apps/wallet-dashboard/app/(protected)/vesting/page.tsx [hunk 5]
```diff
@@ -142,81 +188,155 @@ function VestingDashboardPage(): JSX.Element {
...
-            </div>
-            <div className="flex w-1/2 flex-col items-center justify-center space-y-4 pt-12">
-                <h1>Staked Vesting</h1>
-                <div className="flex flex-row space-x-4">
-                    <div className="flex flex-col items-center rounded-lg border p-4">
-                        <span>Your stake</span>
-                        <span>{vestingSchedule.totalStaked}</span>
+            </Panel>
+            <Panel>
+                <Title title="Staked Vesting" />
```
Comment (evavirseda) [inline apps/wallet-dashboard/app/(protected)/vesting/page.tsx]: missing button for staking :pray: 
image

File: apps/wallet-dashboard/app/(protected)/vesting/page.tsx [hunk 6]
```diff
@@ -142,81 +188,155 @@ function VestingDashboardPage(): JSX.Element {
...
                     </div>
-                )}
-            </div>
+                </div>
+            </Panel>
         </div>
     );
 }
 
+interface TimelockedStakedObjectProps {
```
Comment (evavirseda) [inline apps/wallet-dashboard/app/(protected)/vesting/page.tsx]: can you move all this component to another file?? :pray:

File: apps/wallet-dashboard/app/(protected)/vesting/page.tsx [hunk 7]
```diff
@@ -142,81 +188,155 @@ function VestingDashboardPage(): JSX.Element {
...
+    const supportingText = (() => {
+        if (timelockedStakedObject.stakes.every((s) => s.status === 'Active')) {
+            return {
+                title: 'Estimated Reward',
+                subtitle: `${estimatedRewardFormatted} ${estimatedRewardSymbol}`,
+            };
+        }
+
+        return {
+            title: 'Stake Request Epoch',
```
Comment (evavirseda) [inline apps/wallet-dashboard/app/(protected)/vesting/page.tsx]: should this be showing up here? in figma, it’s not like that, but maybe I’m missing something? :pray: 
image

**Review by evavirseda — DISMISSED** [review]
and conflicts :pray: :pray:

File: apps/wallet-dashboard/lib/utils/vesting/vesting.ts [hunk 8]
```diff
@@ -179,6 +180,12 @@ export function getVestingOverview(
         0,
     );
 
+    const totalEarned = timelockedStakedObjects
+        .filter((t) => t.status === 'Active')
+        .reduce((acc, current) => {
+            return acc + Number(current.estimatedReward);
```
Comment (begonaalvarezd) [inline apps/wallet-dashboard/lib/utils/vesting/vesting.ts]: lets better use `BigInt` here, we should be using it too in https://github.com/iotaledger/iota/blob/46d52c3a5e4ddb57b9779e9e2830fe779c193185/apps/wallet-dashboard/lib/utils/vesting/vesting.ts#L179

File: apps/wallet-dashboard/app/(protected)/vesting/page.tsx [hunk 9]
```diff
@@ -233,143 +248,204 @@ function VestingDashboardPage(): JSX.Element {
...
+                            <VestingScheduleDialog
+                                open={isVestingScheduleDialogOpen}
+                                setOpen={setIsVestingScheduleDialogOpen}
+                                vestingPortfolio={vestingPortfolio}
+                            />
+                        )}
+                    </div>
+                </Panel>
+            </div>
+            <div className="flex w-1/2">
```
Comment (begonaalvarezd) [inline apps/wallet-dashboard/app/(protected)/vesting/page.tsx]: I would use this PR to add the logic to display certain elements on not based on the user's portfolio, so the idea is that, when there user **has no** timelock stakes, this is what is displayed in the screen: vesting & CTA
image
and if the user does have, then the view looks like this (no CTA and vested staking shows on the right) 
image

File: apps/wallet-dashboard/app/(protected)/vesting/page.tsx [hunk 10]
```diff
@@ -233,143 +248,204 @@ function VestingDashboardPage(): JSX.Element {
...
+                {istimelockedStakedObjectsLoading ? (
```
Comment (begonaalvarezd) [inline apps/wallet-dashboard/app/(protected)/vesting/page.tsx]: I would add this loading as something generic in the view to avoid flashes, until the whole vesting schedule and timelocked stakes are loaded

File: apps/wallet-dashboard/app/(protected)/vesting/page.tsx (hunk 7 above)
Comment (begonaalvarezd) [reply apps/wallet-dashboard/app/(protected)/vesting/page.tsx]: yes, if it didnt start earning yet we must display when it will start earning, like we have in the wallet and the wallet dashboard, you can take inspiration from `apps/core/src/components/stake/StakedCard.tsx` https://github.com/iotaledger/iota/blob/46d52c3a5e4ddb57b9779e9e2830fe779c193185/apps/core/src/components/stake/StakedCard.tsx#L72

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]

**Review by brancoder — CHANGES_REQUESTED** [review]
lint 🙏

**Review by brancoder — CHANGES_REQUESTED** [review]
should we move cards in coulmn on smaller devices?
image

also currently there seams to be an issue with cards going under the side bar on smaller viewports
image

**Review by brancoder — APPROVED** [review]
LGTM 🎊 🐼

File: apps/wallet-dashboard/components/staked-timelock-object/StakedTimelockObject.tsx [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+        timelockedStakedObject.validatorAddress;
+
+    // TODO probably we could calculate estimated reward on grouping stage.
+    const summary = timelockedStakedObject.stakes.reduce(
+        (acc, stake) => {
+            const estimatedReward = stake.status === 'Active' ? stake.estimatedReward : 0;
+
+            return {
+                principal: Number(stake.principal) + acc.principal,
+                estimatedReward: Number(estimatedReward) + acc.estimatedReward,
```
Comment (begonaalvarezd) [inline apps/wallet-dashboard/components/staked-timelock-object/StakedTimelockObject.tsx]: these should be bigints if im not wrong https://github.com/iotaledger/iota/blob/2ca2b7bce533cdae5a471a1ef13b45d20f8583a3/crates/iota-framework/packages/iota-system/sources/timelocked_staking.move#L174-L175, https://github.com/iotaledger/iota/blob/2ca2b7bce533cdae5a471a1ef13b45d20f8583a3/crates/iota-framework/packages/iota-system/sources/timelocked_staking.move#L259. I believe the same applies to estimatedReward, you can see how we have bigints in other places in the code https://github.com/iotaledger/iota/blob/2ca2b7bce533cdae5a471a1ef13b45d20f8583a3/apps/core/src/hooks/stake/useTotalDelegatedRewards.ts#L12

File: apps/wallet-dashboard/lib/utils/vesting/vesting.spec.ts [hunk 12]
```diff
@@ -172,11 +183,13 @@ describe('vesting overview', () => {
...
         const lastPayoutValue = Number(lastPayout.principal);
-        const totalAmount =
+        const totalAmount = bigIntRound(
             (SUPPLY_INCREASE_STAKER_VESTING_DURATION *
                 SUPPLY_INCREASE_VESTING_PAYOUTS_IN_1_YEAR *
                 lastPayoutValue) /
-            0.9;
+                0.9,
+        );
+        console.log(totalAmount);
```
Comment (begonaalvarezd) [inline apps/wallet-dashboard/lib/utils/vesting/vesting.spec.ts]: remove

File: apps/wallet-dashboard/lib/utils/vesting/vesting.ts [hunk 13]
```diff
@@ -149,59 +149,69 @@ export function getVestingOverview(
...
+            availableClaiming: 0n,
+            availableStaking: 0n,
         };
     }
 
     const userType = getSupplyIncreaseVestingUserType([latestPayout]);
     const vestingPayoutsCount = getSupplyIncreaseVestingPayoutsCount(userType!);
     // Note: we add the initial payout to the total rewards, 10% of the total rewards are paid out immediately
-    const totalVestedAmount = (vestingPayoutsCount * latestPayout.amount) / 0.9;
+    const totalVestedAmount = BigInt(Math.floor((vestingPayoutsCount * latestPayout.amount) / 0.9));
```
Comment (begonaalvarezd) [inline apps/wallet-dashboard/lib/utils/vesting/vesting.ts:165]: why do we floor?

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]

File: apps/wallet-dashboard/lib/utils/vesting/vesting.ts (hunk 13 above)
Comment (panteleymonchuk) [reply apps/wallet-dashboard/lib/utils/vesting/vesting.ts:165]: bigint doesn't allow not integer to the constructor.

**Review by begonaalvarezd — APPROVED** [review]
